contracts/proxy/transparent/TransparentUpgradeableProxy.sol

```diff
@@ -37,6 +62,9 @@ contract TransparentUpgradeableProxy is ERC1967Proxy {
 
     /**
      * @dev Modifier used internally that will delegate the call to the implementation unless the sender is the admin.
+     *
+     * CAUTION: This modifier is deprecated, as it could cause issues if the modified function has arguments, and the
+     * implementation provides a function with the same selector.
      */
     modifier ifAdmin() {
```

Comment:
Can this just?

```solidity
...
{ else {
  super._fallback();
}
```

We know `msg.sender != _getAdmin()` so the `_fallback` will check that again

Reply:
Using `super._fallback` here has the usual problem we've discussed lately that it may skip an override.

I wouldn't worry about the inefficiency given that the modifier is being deprecated.

Reply:
Gotcha! Thanks